apps/core/lib/core/snitch/data/model/line_item.ex

```diff
@@ -4,7 +4,7 @@ defmodule Core.Snitch.Data.Model.LineItem do
   """
   use Core.Snitch.Data.Model
 
-  @spec get(map()) :: Schema.LineItem.t() | nil | no_return
+  @spec get(map) :: Schema.LineItem.t() | nil | no_return
```

Comment:
I didn't get what  the `no_return` does here, can you elaborate a bit.
And I don't get how are letting the `nil` reach till the `get` of Model, shouldn't it be filtered earlier. or not let it reach the db itself.
Think about it, a db query which searches `nil` and then returns nothing.
I am open to any thoughts on this ?
cc: @ashish173  @pkrawat1

Reply:
Pls read [Ecto.Repo](https://hexdocs.pm/ecto/Ecto.Repo.html#c:get/3) specs. `no_return` is for dialyzer. There's nothing wrong when we return `nil`. If there is no such record in the DB, you might want to return `:error` or `{:error, :not_found}`:
1. Both `:error` and `nil` are atoms, so it makes no difference. `nil` is clearly better.
2. Matching on `nil` is much easier than `{:error, :not_found}`, moreover the only "error" a get might encounter is "not_found", and there's no need to explicitly pass that atom around.